TextToSpeech/impl/TTSSpeaker.cpp

```diff
@@ -1002,6 +1094,11 @@ void TTSSpeaker::GStreamerThreadFunc(void *ctx) {
             speaker->speakText(*data.client->configuration(), data);
         }
 
+        if(!speaker->m_flushed && speaker->m_remoteError) {
```

Comment:
It would be helpful if we add a comment saying this iteration will use local endpoint because of the m_remoteError 

i.e speakText() -> constructURL (this makes use of m_remoteError to determine if local endpoint should be used) -> pipeline reconstruction